src/utils/is.js

```diff
@@ -56,3 +56,29 @@ export const isAbortSignal = object => {
 		)
 	);
 };
+
+/**
+ * isDomainOrSubdomain reports whether sub is a subdomain (or exact match) of
+ * the parent domain.
+ *
+ * Both domains must already be in canonical form.
+ * @param {string|URL} sub
+ * @param {string|URL} parent
+ */
+export const isDomainOrSubdomain = (sub, parent) => {
+	const a = new URL(sub).hostname;
+	const b = new URL(parent).hostname;
+
+	if (a === b) {
+		return true;
+	}
+
+	// If sub is "foo.example.com" and parent is "example.com",
+	// that means sub must end in "."+parent.
+	// Do it without allocating.
+	if (!a.endsWith(b)) {
+		return false;
+	}
+
+	return a[a.length - b.length - 1] === '.';
```

Comment:
What do you think about adding the `.` to the `endsWith` call?

```suggestion
	return a.endsWith(`.${b}`)
```

I think that this is much easier to read ☺️

Reply:
if you are making a request to `fridas-blog.uk.com` and it redirects to `uk.com` then `uk.com` should not know about the cookie...

`a` = referer (original request)
`b` = destination (Location)

Reply:
if you are only after something that is sorter and dose the same thing:

```js
export const isDomainOrSubdomain = (a, b) => {
  a = new URL(a).hostname
  b = new URL(b).hostname
  return a === b || (
    [a.length - b.length - 1] === '.' && a.endsWith(b)
  )
}
```

Reply:
I still don't understand how the two checks are different? As far as I can tell, unless I'm missing something, the two functions below behave the _exact_ same way?

```js
function one (a, b) {
  return a === b || (
    a[a.length - b.length - 1] === '.' && a.endsWith(b)
  )
}

function two (a, b) {
  return a === b || a.endsWith(`.${b}`)
}
```

-----

The case you are mentioning seems to be handled wrong in the committed code then?

```
> isDomainOrSubdomain('http://uk.com', 'http://fridas-blog.uk.com')
true
```

(extra ping @jimmywarting since this is already merged)

Reply:
hmm, double checked again and you are right... they are equally the same, But there is no mistake in the merged code...

I was blinded by the order of arguments passed down to `isDomainOrSubdomain` and mixing up whats gets passed down to the function and in which order. i was just so blinded by how Go and [follow-redirects](https://github.com/follow-redirects/follow-redirects/blob/2ede36d7c60d3acdcd324dcd99a9dbd52e4fb3a6/index.js#L562) also did it...

But at least there is no rush to change it now... can change it later

Reply:
a & b was a stupid variable name...

Reply:
Nice 👍 

Submitted PR to make the code easier to understand here: #1455